src/base64-inl.h

```diff
@@ -182,6 +211,96 @@ inline size_t base64_encode(const char* src,
   return dlen;
 }
 
+
+#if (defined(__x86_64) || defined(__x86_64__)) && \
+    (defined(__GNUC__) || defined(__GNUC))
+#pragma GCC target("avx512vl", "avx512vbmi")
+#include <immintrin.h>
+inline size_t base64_encode_avx512vl(const char* src,
+                                     size_t slen,
+                                     char* dst,
+                                     size_t dlen,
+                                     Base64Mode mode) {
+  // Do the exact check as scalar algorithm.
+  CHECK(dlen >= base64_encoded_size(slen, mode) &&
+        "not enough space provided for base64 encode");
+  size_t dlen_remain = dlen;
+  const char* lookup_tbl = base64_select_table(mode);
+  // 32-bit input
+  // [ 0  0  0  0  0  0  0  0|c1 c0 d5 d4 d3 d2 d1 d0|
+  //  b3 b2 b1 b0 c5 c4 c3 c2|a5 a4 a3 a2 a1 a0 b5 b4]
+  // output order  [1, 2, 0, 1]
+  // [b3 b2 b1 b0 c5 c4 c3 c2|c1 c0 d5 d4 d3 d2 d1 d0|
+  //  a5 a4 a3 a2 a1 a0 b5 b4|b3 b2 b1 b0 c3 c2 c1 c0]
+
+  const __m512i shuffle_input = _mm512_setr_epi32(
+      0x01020001, 0x04050304, 0x07080607, 0x0a0b090a,
+      0x0d0e0c0d, 0x10110f10, 0x13141213, 0x16171516,
+      0x191a1819, 0x1c1d1b1c, 0x1f201e1f, 0x22232122,
+      0x25262425, 0x28292728, 0x2b2c2a2b, 0x2e2f2d2e);
+  const __m512i lookup = _mm512_loadu_si512((const __m512i*)(lookup_tbl));
```

Comment:
Not a safe cast due to possible misalignment. Also due to aliasing but that's less of a concern because node is built with `-fno-strict-aliasing`.

Reply:
> Not a safe cast due to possible misalignment. Also due to aliasing but that's less of a concern because node is built with `-fno-strict-aliasing`.

__mm512_loadu_si512() is unligned load intrinsic so it does not need the data to be aligned.

Reply:
> I think this can/should just be

Yes, I have modfied it as char*